src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataLatestIT.java

```diff
@@ -1694,4 +1694,172 @@ public void testUpdateLocatorsCopyUnsupported() throws SQLException {
       assertFalse(metaData.locatorsUpdateCopy());
     }
   }
+
+  @Test
+  public void testNoPatternSearchAllowedForPrimaryAndForeignKeys() throws SQLException {
+    Map<String, String> params = getConnectionParameters();
+    Properties properties = new Properties();
+    for (Map.Entry<?, ?> entry : params.entrySet()) {
+      if (entry.getValue() != null) {
+        properties.put(entry.getKey(), entry.getValue());
+      }
+    }
+    // test out connection parameter stringsQuoted to remove strings from quotes
+    properties.put("enablePatternSearch", "false");
+    Connection connection = DriverManager.getConnection(params.get("uri"), properties);
```

Comment:
can we use here `getConnection(properties)` method available for integration tests with passing only the parameter we are changing? all other connection parameters should should be set in this method